example-document/03-troubleshooting.md

```diff
@@ -0,0 +1,83 @@
+4. Find out what job is failing
+5. See detailed logs of failed job
+6. Seach for commonly known error messages using native GitHub log search on top right corner
+
+## Validate YAML documents
+
+YAML formatting is strict, and all errors in this job are related to it. Search for `yaml.scanner.ScannerError` to see what went wrong. Example of the log:
+
+```
+yaml.scanner.ScannerError: mapping values are not allowed in this context
+  in "/__w/istqb-ctal-ta/istqb-ctal-ta/sample-exam-answers/questions.yml", line 562, column 19
+```
+To resolve these issues:
+1. Navigate to affected file (`sample-exam-answers/questions.yml` line 562, position 19 in this case)
+2. Resolve the formatting issue which could be cause by several things. The most common are:
+  * wrong indents - fix by adding 2 spaces per indent level
+  * using colon `:` or hashtag `#` in the text - fix by adding the text in quotes `"` or `'`
+  * missing pipe `|` in right after the key name to allow multiline inputs - fix by adding pipe `|` before multiline markdown input
+
+## Produce PDF documents
+
+### Undefined references and citations
+
+The most common failure is one related to references. Search for `Undefined refs and citations` to see what went wrong. Example of the log:
+
+```
+Latexmk: ====Undefined refs and citations with line #s in .tex file:
+  Reference `section:analyze-sut-to-determine-the-appropriate-tas' on page 21 undefined on input line 37
+  Reference `section:apply-layering-of-taf' on page 22 undefined on input line 38
+  Reference `section:apply-design-principles-and-design-patterns-in-ta' on page 24 undefined on input line 38
+  Reference `section:apply-different-approaches-for-automating-tcs' on page 47 undefined on input line 43
+  Reference `section:apply-layering-of-taf' on page 48 undefined on input line 43
+  Reference `section:explain-which-factors-support-and-affect-tas-maintainability' on page 50 undefined on input line 43
+```
+
+Example of the log:
+
+1. Make sure your section identifier `{#apply-layering-of-taf}` is existent within your MD files. See <#section:section-references> for more details.
+2. Make sure you are using the reference in text like `<#section:#apply-layering-of-taf>` correctly. Typos are common here. See <#section:section-references> for more details.
+3. Make sure, all Markdown files you are referencing to are part of the `.tex` file, so they are included in a final rendered file. See <#section:architecture-of-the-solution> for mode details.
+
+
+## Invalid references to files in `syllabus.tex` file
+
+Every MD files you want to include in your document have to be added to TEX file. If the file referenced in the TEX file is not existent (wrong path, wrong name of the file, missing extension, etc.) the pipeline job will fail. Example of the log:
+
+```
+! Package markdown Error: Markdown file
+(markdown)               
+syllabus/01-tasks-in-**test**-process.md
+(markdown)                does not exist
+```
```

Comment:
````suggestion
```
! Package markdown Error: Markdown file
(markdown)                syllabus/01-tasks-in-**test**-process.md
(markdown)                does not exist
```
````

Reply:
Fixed